src/v3/src/util/theme.ts

```diff
@@ -10,116 +10,429 @@
+    set(theme, 'typography.h5.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h5.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h5.fontSize', mergedTokens.TypographySizeHeading5);
+    set(theme, 'typography.h5.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h5.lineHeight', mergedTokens.TypographyLineHeightHeading5);
+    set(theme, 'typography.h6.color', mergedTokens.TypographyColorHeading);
+    set(theme, 'typography.h6.fontFamily', mergedTokens.TypographyFamilyHeading);
+    set(theme, 'typography.h6.fontSize', mergedTokens.TypographySizeHeading6);
+    set(theme, 'typography.h6.fontWeight', mergedTokens.TypographyWeightHeading);
+    set(theme, 'typography.h6.lineHeight', mergedTokens.TypographyLineHeightHeading6);
+    set(theme, 'typography.kbd.fontFamily', mergedTokens.TypographyFamilyMono);
+    set(theme, 'typography.overline.lineHeight', mergedTokens.TypographyLineHeightOverline);
+    set(theme, 'typography.subtitle1.fontFamily', mergedTokens.TypographyFamilyBody);
+    set(theme, 'typography.subtitle2.fontFamily', mergedTokens.TypographyFamilyBody);
+  }
+  return mergeThemes(theme, {
+    components: {
+      MuiInputLabel: {
+        styleOverrides: {
+          root: {
+            wordBreak: 'break-word',
+            whiteSpace: 'normal',
+          },
+        },
+      },
+      MuiButton: {
+        defaultProps: {
+          variant: "primary",
+          disableElevation: true,
+        },
+        styleOverrides: {
+          // @ts-expect-error use type for styleOverride function
+          root: ({ ownerState }) => ({
+            minWidth: "unset",
+            paddingBlock: mergedTokens.Spacing3,
+            paddingInline: mergedTokens.Spacing4,
+            display: "inline-flex",
+            position: "relative",
+            marginBlock: "0",
+            marginInline: "0",
+            transitionProperty: "color, background-color, border-color, box-shadow",
+            transitionDuration: "100ms",
+            transitionTimingFunction: "linear",
+            borderWidth: mergedTokens.BorderWidthMain,
+            borderStyle: mergedTokens.BorderStyleMain,
+            borderRadius: mergedTokens.BorderRadiusMain,
+            borderColor: "transparent",
+            fontSize: mergedTokens.TypographySizeBody,
+            fontWeight: mergedTokens.TypographyWeightBodyBold,
+            fontFamily: mergedTokens.TypographyFamilyButton,
+            lineHeight: mergedTokens.TypographyLineHeightUi,
+            whiteSpace: "nowrap",
 
-  odysseyTheme.direction = languageDirection;
+            [`.${buttonClasses.root} + &`]: {
+              marginInlineStart: mergedTokens.Spacing2,
+            },
 
-  // Do not modify `odysseyTheme` after this line, it will have no effect on the resulting theme
-  const odysseyThemeCopy = cloneDeep(odysseyTheme);
```

Comment:
AFAICT there was no difference cloning or modifying the `odysseyTheme` -- lmk if there's something I'm missing. Deep-cloning is expensive so removed.